ai_transform/engine/abstract_engine.py

```diff
@@ -418,13 +418,8 @@ def name(self, value):
         self._name = value
 
     def set_success_ratio(self) -> None:
-        try:
-            self._success_ratio = self._successful_documents / (
-                self.size * len(self.operators)
-            )
-        except ZeroDivisionError:
-            # if theres no documents just set success ratio at 1
-            self._success_ratio = 1
+        denominator = max((self.size * len(self.operators)), 1)
+        self._success_ratio = self._successful_documents / denominator
```

Comment:
I disagree with this behavior. 
If number of documents is 0 then the workflow should stil be successful.